File: backend/schemas/market_overview_split_trends.py
"""Schemas for split-trend analysis on market overview."""

from pydantic import BaseModel


class SplitTrendPoint(BaseModel):
    period: str
    value: float | None = None


class SplitTrendSeries(BaseModel):
    label: str
    data: list[SplitTrendPoint]


class SplitTrendResponse(BaseModel):
    metric: str
    metric_label: str
    split_by: str
    split_by_label: str
    title: str
    subtitle: str
    splits: list[SplitTrendSeries]

File: frontend-next/src/types/marketOverviewSplitTrends.ts
export interface SplitTrendPoint {
  period: string;
  value: number | null;
}

export interface SplitTrendSeries {
  label: string;
  data: SplitTrendPoint[];
}

export interface SplitTrendResponse {
  metric: string;
  metric_label: string;
  split_by: string;
  split_by_label: string;
  title: string;
  subtitle: string;
  splits: SplitTrendSeries[];
}

export interface SplitTrendOption {
  value: string;
  label: string;
}

export interface SplitTrendOptionsResponse {
  metrics: SplitTrendOption[];
  split_by: SplitTrendOption[];
}
